integrationTests/multiShard/smartContract/scCallingSC_test.go

```diff
@@ -511,6 +512,155 @@ func TestSCCallingInCrossShardDelegation(t *testing.T) {
 	}
 }
 
+func TestSCCallingInCrossShardDelegation(t *testing.T) {
```

Comment:
This test and the test `TestSCCallingInCrossShardDelegationMock()` have a very similar set-up part. Please extract the common code as functions.